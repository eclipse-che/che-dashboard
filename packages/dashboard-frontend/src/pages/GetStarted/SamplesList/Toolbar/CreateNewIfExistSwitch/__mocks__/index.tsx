/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '..';

export class CreateNewIfExistSwitch extends React.PureComponent<Props> {
  private isChecked = false;

  render(): React.ReactNode {
    return (
      <div data-testid="create-new-if-exist">
        {this.isChecked ? 'Create New If Existing On' : 'Create New If Existing Off'}
        <input
          data-testid="create-new-if-exist-switch"
          type="checkbox"
          onClick={() => {
            this.isChecked = !this.isChecked;
            this.props.onChange(this.isChecked);
          }}
          name="Toggle Create New If Existing"
        />
      </div>
    );
  }
}
