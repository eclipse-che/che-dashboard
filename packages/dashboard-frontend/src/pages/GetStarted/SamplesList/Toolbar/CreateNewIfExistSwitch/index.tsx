/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { Switch } from '@patternfly/react-core';
import React from 'react';

export type Props = {
  onChange: (isCreateNewIfExist: boolean) => void;
};
type State = {
  isChecked: boolean;
};

export class CreateNewIfExistSwitch extends React.PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      isChecked: false, // Default state is unchecked
    };
  }

  private handleChange(isChecked: boolean): void {
    this.setState({ isChecked });
    this.props.onChange(isChecked);
  }

  private getLabelNode(text: string): React.ReactNode {
    return <div style={{ minWidth: '200px' }}>{text}</div>;
  }

  render(): React.ReactElement {
    const { isChecked } = this.state;
    // Using getLabelNode to ensure consistent styling
    const labelOn = this.getLabelNode('Create New If Existing On');
    const labelOff = this.getLabelNode('Create New If Existing Off');

    return (
      <Switch
        id="create-new-if-exist-switch"
        label={labelOn}
        labelOff={labelOff}
        isChecked={isChecked}
        onChange={isChecked => this.handleChange(isChecked)}
      />
    );
  }
}
