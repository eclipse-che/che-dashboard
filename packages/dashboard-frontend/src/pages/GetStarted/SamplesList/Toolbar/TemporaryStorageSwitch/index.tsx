/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { Switch, Text, Tooltip, TooltipPosition } from '@patternfly/react-core';
import { OutlinedQuestionCircleIcon } from '@patternfly/react-icons';
import React from 'react';
import { connect, ConnectedProps } from 'react-redux';

import { RootState } from '@/store';
import { selectBranding } from '@/store/Branding/selectors';

export type Props = MappedProps & {
  isTemporary: boolean;
  onChange: (isTemporary: boolean) => void;
};
type State = {
  isChecked: boolean;
};

class TemporaryStorageSwitch extends React.PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      isChecked: this.props.isTemporary,
    };
  }

  private handleChange(isChecked: boolean): void {
    this.setState({ isChecked });
    this.props.onChange(isChecked);
  }

  private getLabelNode(text: string): React.ReactNode {
    return <div style={{ minWidth: '170px' }}>{text}</div>;
  }

  render(): React.ReactElement {
    const { branding } = this.props;
    const { isChecked } = this.state;

    // Using getLabelNode to ensure consistent styling
    const labelOn = this.getLabelNode('Temporary Storage On');
    const labelOff = this.getLabelNode('Temporary Storage Off');

    const storageTypesDocLink = branding.docs.storageTypes;

    return (
      <React.Fragment>
        <Switch
          id="temporary-storage-switch"
          label={labelOn}
          labelOff={labelOff}
          isChecked={isChecked}
          onChange={isChecked => this.handleChange(isChecked)}
          aria-describedby="temporary-storage-tooltip"
        />
        <span style={{ marginLeft: '10px' }}>
          <Tooltip
            id="temporary-storage-tooltip"
            isContentLeftAligned={true}
            position={TooltipPosition.top}
            content={
              <React.Fragment>
                Temporary Storage allows for faster I/O but may have limited storage and is not
                persistent.
                <Text>
                  <a rel="noreferrer" target="_blank" href={storageTypesDocLink}>
                    Open documentation page
                  </a>
                </Text>
              </React.Fragment>
            }
          >
            <OutlinedQuestionCircleIcon />
          </Tooltip>
        </span>
      </React.Fragment>
    );
  }
}

const mapStateToProps = (state: RootState) => ({
  branding: selectBranding(state),
});

const connector = connect(mapStateToProps);

type MappedProps = ConnectedProps<typeof connector>;

export default connector(TemporaryStorageSwitch);
