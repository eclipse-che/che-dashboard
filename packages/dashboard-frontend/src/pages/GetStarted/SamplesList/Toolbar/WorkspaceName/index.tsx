/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { FormGroup, TextInput, ValidatedOptions } from '@patternfly/react-core';
import { ExclamationCircleIcon } from '@patternfly/react-icons';
import isEqual from 'lodash/isEqual';
import React from 'react';
import { connect, ConnectedProps } from 'react-redux';

import { CheTooltip } from '@/components/CheTooltip';
import styles from '@/pages/GetStarted/SamplesList/Toolbar/WorkspaceName/index.module.css';
import { RootState } from '@/store';
import { selectAllWorkspaces } from '@/store/Workspaces';

const MAX_LENGTH = 63;
const ERROR_MAX_LENGTH = 'The name is not valid.';
const ERROR_TOOLTIP_MAX_LENGTH = `The maximum length is ${MAX_LENGTH} characters.`;
const PATTERN = `^(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?$`;
const ERROR_PATTERN_MISMATCH = 'The name is not valid.';
const ERROR_TOOLTIP_PATTERN_MISMATCH =
  'The name is invalid. It can contain only digits, latin letters, scores, underscores. It must start and end with a letter or digit and cannot include special characters such as spaces or symbols (e.g., $, @, #, etc.).';
const ERROR_PATTERN_EXISTING_NAME = 'The name is already in use.';
const ERROR_TOOLTIP_PATTERN_EXISTING_NAME =
  'The name is already in use. Please choose another name.';

export type Props = MappedProps & {
  onChange: (name: string) => void;
  callbacks?: {
    reset?: () => void;
  };
};

export type State = {
  workspaceName: string;
  errorMessage?: string;
  errorTooltipMessage?: string;
  validated: ValidatedOptions;
};

class WorkspaceNameFormGroup extends React.PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      workspaceName: '',
      validated: ValidatedOptions.default,
    };
  }

  public componentDidUpdate(prevProps: Readonly<Props>): void {
    if (!isEqual(this.props.allWorkspaces, prevProps.allWorkspaces)) {
      const { workspaceName } = this.state;
      const { validated, errorMessage, errorTooltipMessage } = this.validate(workspaceName);
      if (
        validated !== this.state.validated ||
        errorMessage !== this.state.errorMessage ||
        errorTooltipMessage !== this.state.errorTooltipMessage
      ) {
        this.setState({ validated, errorMessage, errorTooltipMessage });
        this.props.onChange(validated === ValidatedOptions.error ? '' : workspaceName);
      }
    }

    if (
      this.props.callbacks !== undefined &&
      this.props.callbacks.reset === undefined &&
      this.state.workspaceName.length > 0
    ) {
      this.props.callbacks.reset = () => {
        this.setState({
          workspaceName: '',
          validated: ValidatedOptions.default,
          errorMessage: undefined,
          errorTooltipMessage: undefined,
        });
        this.handleWorkspaceNameChange('');
      };
    }
  }

  private handleWorkspaceNameChange(workspaceName: string): void {
    const { validated, errorMessage, errorTooltipMessage } = this.validate(workspaceName);
    this.setState({ workspaceName, validated, errorMessage, errorTooltipMessage });

    this.props.onChange(validated === ValidatedOptions.error ? '' : workspaceName);
  }

  private validate(name: string): {
    errorMessage: string | undefined;
    errorTooltipMessage: string | undefined;
    validated: ValidatedOptions;
  } {
    if (name.length === 0) {
      return {
        errorMessage: undefined,
        errorTooltipMessage: undefined,
        validated: ValidatedOptions.default,
      };
    } else if (name.length > MAX_LENGTH) {
      return {
        errorMessage: ERROR_MAX_LENGTH,
        errorTooltipMessage: ERROR_TOOLTIP_MAX_LENGTH,
        validated: ValidatedOptions.error,
      };
    }
    if (this.props.allWorkspaces.some(w => name === w.name || name === w.ref.metadata.name)) {
      return {
        errorMessage: ERROR_PATTERN_EXISTING_NAME,
        errorTooltipMessage: ERROR_TOOLTIP_PATTERN_EXISTING_NAME,
        validated: ValidatedOptions.error,
      };
    }
    if (!new RegExp(PATTERN).test(name)) {
      return {
        errorMessage: ERROR_PATTERN_MISMATCH,
        errorTooltipMessage: ERROR_TOOLTIP_PATTERN_MISMATCH,
        validated: ValidatedOptions.error,
      };
    }

    return {
      errorMessage: undefined,
      errorTooltipMessage: undefined,
      validated: ValidatedOptions.success,
    };
  }

  private getHelperTextInvalid(): React.ReactNode {
    const { validated, errorMessage, errorTooltipMessage } = this.state;

    if (validated !== ValidatedOptions.error) {
      return undefined;
    }

    if (errorTooltipMessage) {
      return (
        <CheTooltip content={errorTooltipMessage}>
          <div className={styles.helperTextInvalid}>{errorMessage}</div>
        </CheTooltip>
      );
    }

    return <div className={styles.helperTextInvalid}>{errorMessage}</div>;
  }

  public render(): React.ReactNode {
    const { workspaceName, validated } = this.state;
    const helperTextInvalid = this.getHelperTextInvalid();

    return (
      <FormGroup
        validated={validated}
        className={styles.inputFormGroup}
        helperTextInvalid={helperTextInvalid}
        helperTextInvalidIcon={<ExclamationCircleIcon />}
      >
        <TextInput
          type="text"
          id="wrks-name-input"
          value={workspaceName}
          validated={validated}
          maxLength={MAX_LENGTH + 1}
          placeholder="Enter a name for the sample workspace (Optional)"
          onChange={name => this.handleWorkspaceNameChange(name)}
        />
      </FormGroup>
    );
  }
}

const mapStateToProps = (state: RootState) => ({
  allWorkspaces: selectAllWorkspaces(state),
});

const connector = connect(mapStateToProps);

type MappedProps = ConnectedProps<typeof connector>;

export default connector(WorkspaceNameFormGroup);
