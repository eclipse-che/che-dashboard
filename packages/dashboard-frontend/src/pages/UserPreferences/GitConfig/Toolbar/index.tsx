/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import {
  Button,
  ButtonVariant,
  Toolbar,
  ToolbarContent,
  ToolbarItem,
} from '@patternfly/react-core';
import { CodeIcon, ListIcon, PlusCircleIcon } from '@patternfly/react-icons';
import React from 'react';

export type Props = {
  mode: 'form' | 'viewer';
  onAdd: () => void;
  onChangeMode: (mode: 'form' | 'viewer') => void;
};

export class GitConfigToolbar extends React.PureComponent<Props> {
  private handleModeChange(mode: 'form' | 'viewer'): void {
    this.setState({ mode });
    this.props.onChangeMode(mode);
  }

  public render(): React.ReactElement {
    const { mode } = this.props;

    return (
      <Toolbar style={{ paddingBottom: 0 }}>
        <ToolbarContent>
          <ToolbarItem alignment={{ default: 'alignRight' }}>
            {mode === 'form' ? (
              <Button
                variant={ButtonVariant.link}
                icon={<CodeIcon />}
                iconPosition="left"
                onClick={() => this.handleModeChange('viewer')}
              >
                Switch to Viewer
              </Button>
            ) : (
              <Button
                variant={ButtonVariant.link}
                icon={<ListIcon />}
                iconPosition="left"
                onClick={() => this.handleModeChange('form')}
              >
                Switch to Form
              </Button>
            )}
            <Button
              variant={ButtonVariant.link}
              icon={<PlusCircleIcon />}
              iconPosition="left"
              onClick={() => this.props.onAdd()}
            >
              Import Git Configuration
            </Button>
          </ToolbarItem>
        </ToolbarContent>
      </Toolbar>
    );
  }
}
