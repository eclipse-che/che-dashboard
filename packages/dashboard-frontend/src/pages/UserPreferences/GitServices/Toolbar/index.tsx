/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import {
  Button,
  ButtonVariant,
  Toolbar,
  ToolbarContent,
  ToolbarItem,
} from '@patternfly/react-core';
import React from 'react';

import { IGitOauth } from '@/store/GitOauthConfig/types';

export type Props = {
  isDisabled: boolean;
  selectedItems: IGitOauth[];
  onRevokeButton: () => void;
};

export class GitServicesToolbar extends React.PureComponent<Props> {
  public handleRevokeButtonClick(): void {
    this.props.onRevokeButton();
  }

  render(): React.ReactNode {
    const { isDisabled, selectedItems } = this.props;

    const buttonDisabled = isDisabled || selectedItems.length === 0;

    return (
      <React.Fragment>
        <Toolbar>
          <ToolbarContent>
            <ToolbarItem>
              <Button
                variant={ButtonVariant.danger}
                isDisabled={buttonDisabled}
                data-testid="bulk-revoke-button"
                onClick={() => this.handleRevokeButtonClick()}
              >
                Revoke
              </Button>
            </ToolbarItem>
          </ToolbarContent>
        </Toolbar>
      </React.Fragment>
    );
  }
}
