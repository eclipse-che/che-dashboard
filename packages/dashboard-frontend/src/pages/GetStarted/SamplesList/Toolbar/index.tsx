/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { Flex, FlexItem, Text, TextContent, TextInput } from '@patternfly/react-core';
import React from 'react';
import Pluralize from 'react-pluralize';
import { connect, ConnectedProps } from 'react-redux';

import TemporaryStorageSwitch from '@/pages/GetStarted/SamplesList/Toolbar/TemporaryStorageSwitch';
import { AppState } from '@/store';
import * as DevfileRegistriesStore from '@/store/DevfileRegistries';
import { selectFilterValue, selectMetadataFiltered } from '@/store/DevfileRegistries/selectors';

export type Props = MappedProps & {
  isTemporary: boolean;
  onTemporaryStorageChange: (isTemporary: boolean) => void;
};

class SamplesListToolbar extends React.PureComponent<Props> {
  componentWillUnmount(): void {
    this.props.clearFilter();
  }

  private handleTextInputChange(searchValue: string): void {
    this.props.setFilter(searchValue);
  }

  private buildCount(foundCount: number, searchValue: string): React.ReactElement {
    return searchValue === '' ? (
      <></>
    ) : (
      <Pluralize singular={'item'} count={foundCount} zero={'Nothing found'} />
    );
  }

  render(): React.ReactElement {
    const { filterValue, isTemporary, metadataFiltered } = this.props;

    const foundCount = metadataFiltered.length;

    return (
      <Flex>
        <FlexItem>
          <TextInput
            value={filterValue}
            type="search"
            onChange={value => this.handleTextInputChange(value)}
            aria-label="Filter samples list"
            placeholder="Filter by"
          />
        </FlexItem>
        <FlexItem>
          <TextContent>
            <Text>{this.buildCount(foundCount, filterValue)}</Text>
          </TextContent>
        </FlexItem>
        <FlexItem align={{ default: 'alignRight' }}>
          <TemporaryStorageSwitch
            isTemporary={isTemporary}
            onChange={isTemporary => this.props.onTemporaryStorageChange(isTemporary)}
          />
        </FlexItem>
      </Flex>
    );
  }
}

const mapStateToProps = (state: AppState) => ({
  filterValue: selectFilterValue(state),
  metadataFiltered: selectMetadataFiltered(state),
});

const connector = connect(mapStateToProps, DevfileRegistriesStore.actionCreators);

type MappedProps = ConnectedProps<typeof connector>;
export default connector(SamplesListToolbar);
