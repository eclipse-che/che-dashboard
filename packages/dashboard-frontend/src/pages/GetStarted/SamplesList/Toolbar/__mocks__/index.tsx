/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/GetStarted/SamplesList/Toolbar';

export default class SamplesListToolbar extends React.PureComponent<Props> {
  render(): React.ReactNode {
    const { isTemporary, onTemporaryStorageChange } = this.props;
    return (
      <div>
        <span>Samples List Toolbar</span>
        <span data-testid="isTemporary">{isTemporary ? 'true' : 'false'}</span>
        <button onClick={() => onTemporaryStorageChange(!isTemporary)}>Toggle isTemporary</button>
      </div>
    );
  }
}
