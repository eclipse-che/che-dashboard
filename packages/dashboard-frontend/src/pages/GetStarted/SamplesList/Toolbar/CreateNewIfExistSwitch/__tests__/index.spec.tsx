/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { screen } from '@testing-library/react';
import React from 'react';

import { CreateNewIfExistSwitch } from '@/pages/GetStarted/SamplesList/Toolbar/CreateNewIfExistSwitch';
import getComponentRenderer from '@/services/__mocks__/getComponentRenderer';

const { renderComponent, createSnapshot } = getComponentRenderer(getComponent);

const mockOnChange = jest.fn();

// mute console.error
console.error = jest.fn();

describe('Create New If Exist Switch', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('snapshot', () => {
    const snapshot = createSnapshot();
    expect(snapshot.toJSON()).toMatchSnapshot();
  });

  it('should be initially switched off', () => {
    renderComponent();
    const switchInput = screen.getByRole('checkbox') as HTMLInputElement;
    expect(switchInput.checked).toBeFalsy();

    switchInput.click();
    expect(switchInput.checked).toBeTruthy();
    expect(mockOnChange).toHaveBeenCalledTimes(1);
  });
});

function getComponent() {
  return <CreateNewIfExistSwitch onChange={mockOnChange} />;
}
