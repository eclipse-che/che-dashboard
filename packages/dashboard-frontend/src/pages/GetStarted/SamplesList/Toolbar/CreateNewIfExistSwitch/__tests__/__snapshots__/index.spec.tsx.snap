// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create New If Exist Switch snapshot 1`] = `
<label
  className="pf-c-switch"
  data-ouia-component-id="OUIA-Generated-Switch-1"
  data-ouia-component-type="PF4/Switch"
  data-ouia-safe={true}
  htmlFor="create-new-if-exist-switch"
>
  <input
    aria-label=""
    aria-labelledby="create-new-if-exist-switch-off"
    checked={false}
    className="pf-c-switch__input"
    disabled={false}
    id="create-new-if-exist-switch"
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="pf-c-switch__toggle"
  />
  <span
    aria-hidden="true"
    className="pf-c-switch__label pf-m-on"
    id="create-new-if-exist-switch-on"
  >
    <div
      style={
        {
          "minWidth": "200px",
        }
      }
    >
      Create New If Existing On
    </div>
  </span>
  <span
    aria-hidden="true"
    className="pf-c-switch__label pf-m-off"
    id="create-new-if-exist-switch-off"
  >
    <div
      style={
        {
          "minWidth": "200px",
        }
      }
    >
      Create New If Existing Off
    </div>
  </span>
</label>
`;
